File: Java/ShellSort.java
import java.util.Arrays;
import java.util.Random;

public class ShellSort {
    static int MAX_VALUE = 101;

    public static void main(String[] args) {
        Random random = new Random();
        int array[] = new int[32];
        for (int i = 0; i < array.length; i++) {
            array[i] = random.nextInt(MAX_VALUE);
        }

        System.out.println(Arrays.toString(array));
        shellSort(array);
        System.out.println(Arrays.toString(array));
    }

    private static void shellSort(int array[]) {
        for (int gap = array.length / 2; gap > 0; gap /= 2) {
            for (int i = gap; i < array.length; i += 1) {
                int temp = array[i];

                int j;
                for (j = i; j >= gap && array[j - gap] > temp; j -= gap)
                    array[j] = array[j - gap];

                array[j] = temp;
            }
        }
    }
}

File: Python/ShellSort.py

import random

def shellSort(array):
    n = len(array)
    gap = n // 2
    while (gap > 0):
        for i in range(gap, n):
            temp = array[i]

            j = i
            while (j >= gap and array[j-gap] > temp):
                array[j] = array[j-gap]
                j -= gap
            array[j] = temp

        gap //= 2
    
def main():
    MAX_VALUE = 100
    ARRAY_LENGTH = 32
    array = [0] * ARRAY_LENGTH    
    for i in range(len(array)):
        array[i] = random.randint(0, 100)
    print(array)
    shellSort(array)
    print(array)
    
if __name__ == "__main__":
    main()